numpy/core/tests/test_regression.py

```diff
@@ -2115,6 +2115,10 @@ def __array__(self,*args,**kwargs):
         for f in [op.lt, op.le, op.gt, op.ge]:
             if sys.version_info[0] >= 3:
                 assert_raises(TypeError, f, lhs, rhs)
+            elif sys.version_info[0] < 3 and sys.py3kwarning:
+                warnings.filterwarnings('error',category=DeprecationWarning)
+                assert_raises(DeprecationWarning, f, lhs, rhs)
+                warnings.filterwarnings('ignore',category=DeprecationWarning)
```

Comment:
Hmm, I see. I also wondered why so many tests in a single file are unexpectedly failing.

On second thought, should I just ignore this test during -3 switch? 
The test will anyhow run without -3 as part of the whole test suite..?

All other comments I'll take care asap